### DISCUSSION TITLE
Development Discussions

### DISCUSSION THREAD

**Discussion by Nonethousand** [0K9090/ScratchLang#1] [Announcements]

**Comment by Steb03** [comment]
Which version should I write about?

**Reply by Nonethousand** [reply]
The Python version of the decompiler. If you don't understand the code, I can explain it for you.

**Reply by Steb03** [reply]
Ok

**Reply by Steb03** [reply]
Also, try to snip out the background from the images, so that it is more on-theme.

**Reply by Nonethousand** [reply]
Ok, I cropped the images.

**Reply by Steb03** [reply]
@0K9090 Sorry for being annoying, but V2, right?

**Reply by Nonethousand** [reply]
Write about scratchlang.py, since it's the current and recommended version. If you don't understand Python I could add some comments to the code.

**Reply by Steb03** [reply]
https://github.com/ScratchLang/scratch-lang-docs/commit/ea2efb5762eb84098047f8a69e594c67c44b8436 Am I writing it correctly so far?

**Reply by Nonethousand** [reply]
@stan-solov Couple of things wrong, there are spaces in between the parenthesis and brackets.
For example, you wrote `wait("n")seconds` but it should be `wait ("n") seconds`

Also, why did you write C-Blocks like `if <INPUT> then {} else {}` instead of
```
if <INPUT> {
} else {
}

**Reply by Steb03** [reply]
Oh, because I wanted to be inline, but I could put the description on the bottom if you'd like

**Reply by Nonethousand** [reply]
Sure.

**Comment by Nonethousand** [comment]
BTW, @stan-solov 
can you invite me to that editor you're making?

I'm a noob at html/css and I don't even know js, but maybe I can help by finding bugs or something.

**Reply by Steb03** [reply]
I think I need gitpod for that.

**Reply by Steb03** [reply]
I use github.dev, so just press the `.` button when your on the repo home page.

**Reply by Nonethousand** [reply]
Ok.

**Reply by Nonethousand** [reply]
> I use github.dev, so just press the `.` button when your on the repo home page.

The homepage of ScratchLang, right?

**Reply by Steb03** [reply]
Of the repo, yes

**Reply by Nonethousand** [reply]
Oh. I kinda thought you were talking about that editor you are making.

**Reply by Steb03** [reply]
Oh, yeah. I kinda stopped working on it because tosh is better and I cant get codemirror to work without NPM.

**Comment by Steb03** [comment]
Hey should we change the org profile img to this?
![favicon](https://user-images.githubusercontent.com/113939594/199379755-2cfd2d9f-4c26-45c6-8ef4-26141962bacd.png)

**Reply by Nonethousand** [reply]
That's really good, how did you make it?

**Reply by Steb03** [reply]
Figma

**Reply by Steb03** [reply]
In like 30 seconds

**Reply by Nonethousand** [reply]
ok, ill make sure to check it out

I made my shitty one with inkscape

**Reply by Steb03** [reply]
(Offtopic)
![image](https://user-images.githubusercontent.com/113939594/199383984-94aa7512-3b37-468e-a890-5e2bc6454c56.png)

**Reply by Nonethousand** [reply]
What is that? Follower count?

**Reply by Steb03** [reply]
Post in ats

**Reply by Steb03** [reply]
It would've been better if you would have downloaded it from here of from [here](https://github.com/ScratchLang/scratch-lang-docs/blob/main/images/favicon.png), but it's fine

**Comment by Nonethousand** [comment]
@stan-solov (only mentioned you because you're the only contributor so far)
I'm making a built-in text editor for ScratchLang, with syntax highlighting. What do you think of it?
![mintty 2022-11-05_18-00-07](https://user-images.githubusercontent.com/78574005/200149137-957679d6-189a-4edf-a3e1-a6e060726a67.png)

**Reply by Steb03** [reply]
I think that is very good! Are you making different modes? (you dont have to, but if you were, that would look like the high-contrast mode that vscode has.)

**Reply by Steb03** [reply]
Also, are you making it on a repo, or some sort of terminal thing?

**Reply by Nonethousand** [reply]
> Also, are you making it on a repo, or some sort of terminal thing?

It's a terminal-based text editor.

**Reply by Steb03** [reply]
Ok

**Comment by Nonethousand** [comment]
@stan-solov 
Added some stuff to the editor, how does it look so far? What should I add?

https://user-images.githubusercontent.com/78574005/202261119-7c8ef881-8696-4e37-b6f7-edae9d9e27f8.mp4

**Reply by Steb03** [reply]
I think it looks great so far! Don't you think it's kinda hard to see what's written in the bright green highlight?

**Reply by Nonethousand** [reply]
The green highlighting supposed to make it easier to see the \nscript and \prep... should I change the color?

**Reply by Steb03** [reply]
Oh, well then it could be useful if you are trying to look for sections of code. You should keep it.

**Comment by Nonethousand** [comment]
Should I keep the editor bundled with ScratchLang, or should I make a new repo for it?

**Reply by Steb03** [reply]
Idk. If you are more of an organized person, I'd make a repo.

**Comment by Nonethousand** [comment]
Hey @veloce-vasta, how is the documentation going so far? I'm thinking about adding some stuff but it might conflict with any changes you might have.

Also, I found that new [discussion thread you made](https://github.com/orgs/ScratchLang/teams/devlopment/discussions/1) so I'll probably use that from now on

**Reply by Steb03** [reply]
No, you can commit

**Comment by wendiner** [comment]
Is this project dead?  
I'd love to work on it but I'm not sure if anybody else is still going.

**Reply by Nonethousand** [reply]
It's not dead. A lot has stuff has happened. Currently I am developing ScratchLang in C++.

**Reply by wendiner** [reply]
Alright, if you say so. I could make a web implementation in JavaScript once the main version is closer to being finished.

Edit: I'm just gonna start working on it right now. I think I have a general idea.

**Reply by Nonethousand** [reply]
Which version are you using as a reference (theres a Python version, C# version, and C++ version with the Python version being the most up-to-date)

**Reply by wendiner** [reply]
I planned on going off of the README, but if there's a physical version that's further along in development, it would probably be a better choice. The Python version, then.

Edit: The Python compiler seems incomplete. Instead of creating other versions, I should probably focus on helping complete the ones already in development.
But in order to do so, I need a clear understanding of ScratchLang's finer details. Mostly syntax.

**Comment by Artificial-IntelligenceAI** [comment]
Lol
